### DISCUSSION TITLE
How should I go about unlocking characters?

### DISCUSSION THREAD

**Discussion by aflacc** [HaxeFlixel/flixel#2504] [Q&A] [answered]
So I want to make a thing where you can unlock characters and then choose them, but I'm not sure what method I should use to actually track what skins are unlocked. Should I have an array or something for each skin or something or use variables for each skin

**Comment by RichardBray** [comment, marked as answer]
This is a good question but I guess more of a general game dev question than a HF specific one. Does your game support saving or not?

If it supports saving then you can just use FlxSave to store an object of data:
```
{
  'characterOneUnlocked': true,
  'characterTwoUnlocked': false,
}
```

I would not suggest having using nested objects in FlxSave as I've had bad results with this, instead, I would say you should use a string of 1s and 0s
```
{
  'unlockedCharacters': '1, 0'
}
``` 
and then turn that into an array.

If your game doesn't support saves then you could store everything in a singleton that all the states in your game would have access to. I have a tutorial on how to do that here, although it's not flixel specific:

https://www.youtube.com/watch?v=vcRCgFcGUzQ

**Reply by aflacc** [reply]
Thanks for the suggestion! I actually found a really good way to handle the skins.
```hx
var skins:Array<Dynamic> = [ // actual charcater name | icon name | display text | locked | size | file reference
		['default','none','Default (None)',true,0.5,'default']
];
```

Works really well because I can assign as many values to each skin i want.
Its really good

Also yes, my game does use saving and I'm still working on the saving system but I have figured out a good way to actually change the values
Basically I'm just making another array which is bools only
